extension/intents/timer/timer.js

```diff
@@ -13,21 +14,27 @@ class TimerController {
     return this.activeTimer;
   }
 
-  closeActiveTimer() {
-    this.activeTimer.close();
-    this.activeTimer = null;
-  }
-
   setActiveTimer(totalInMS, context) {
     this.activeTimer = new Timer(totalInMS, context);
     this.lastActiveTimer = { ...this.activeTimer };
     this.activeTimer.start();
+    this.waitPromise = new Promise((resolve, reject) => {
+      this.closeActiveTimer = () => {
+        this.activeTimer.close();
+        this.activeTimer = null;
+        resolve(true);
+      };
+    });
```

Comment:
You could also use `util.makeNakedPromise()` here to make the logic slightly simpler (then `closeActiveTimer()` could just call `this.waitPromise.resolve(true)`).

Reply:
Is closeActiveTimer really going to work? It seems like it's only called by `"timer.close"`, and not by the timer actually running out...? As far as I can tell, when the timer runs out it opens the popup but doesn't call this close method.